charlotte_core/src/acpi/rsdp.rs

```diff
@@ -0,0 +1,108 @@
+use core::str;
+
+const RSDP_SIGNATURE: [u8; 8] = *b"RSD PTR ";
+
+const RDSP_V1_LEN: usize = 20; // Length of the RSDP for version 1 (20 bytes)
+
+/// Contains information about the RSDP (Root System Description Pointer).
+#[repr(C, packed)]
+#[derive(Copy, Clone)]
+pub struct Rsdp {
+    signature: [u8; 8],
+    checksum: u8,
+    oem_id: [u8; 6],
+    revision: u8,
+    rsdt_address: u32,
+    // Only valid on revison 2.0 and later
+    length: u32,
+    xsdt_address: u64,
+    extended_checksum: u8,
+    reserved: [u8; 3],
+}
+
+impl Rsdp {
+    /// Creates a new RSDP from an address
+    pub fn new_from_address(address: usize) -> Self {
+        let rsdp = unsafe { &*(address as *const Rsdp) };
+        if rsdp.validate() {
+            *rsdp
+        } else {
+            panic!("Failed to validate RSDP.")
```

Comment:
Maybe add to the log if the checksum failed or the string is not valid utf8 (might happen with some weird bios or am I being paranoid?

Reply:
I second this. We want to be able to know as much about the cause of a kernel panic as possible for diagnostic purposes. In this case we would know that the issue is on the firmware side and not ours.